src/6_branch.js

```diff
@@ -1027,6 +1016,187 @@ Branch.prototype['sendSMS'] = wrap(
+ * );
+ * ```
+ *
+ * #### Example
+ * ```js
+ * branch.deepview(
+ *     {
+ *         channel: 'facebook',
+ *         data: {
+ *             mydata: 'content of my data',
+ *             foo: 'bar',
+ *             '$desktop_url': 'https://en.wikipedia.org/wiki/Internet',
+ *             '$ios_url': 'https://en.wikipedia.org/wiki/Internet',
+ *             '$ipad_url': 'https://en.wikipedia.org/wiki/Internet',
+ *             '$android_url': 'https://en.wikipedia.org/wiki/Internet',
+ *             '$deepview_path': 'item_id=12345',
+ *             '$og_app_id': '12345',
+ *             '$og_title': 'My App',
+ *             '$og_description': 'My app\'s description.'
+ *         },
+ *         feature: 'dashboard',
+ *         stage: 'new user',
+ *         tags: [ 'tag1', 'tag2' ],
+ *     },
+ *     {
+ *         make_new_link: true,
+ *         open_app: true
+ *     },
+ *     function(err) {
+ *         console.log(err || 'no error');
+ *     }
+ * );
+ * ```
+ *
+ * ##### Callback Format
+ * ```js
+ * callback(
+ *     "Error message"
+ * );
+ * ```
+ *
+ */
+/*** +TOC_ITEM #deepviewdata-options-callback &.deepview()& ^ALL ***/
+Branch.prototype['deepview'] = wrap(
+	callback_params.CALLBACK_ERR,
+	function(done, data, options) {
+		var self = this;
+
+		if (!options) {
+			options = { };
+		}
+		var cleanedData = utils.cleanLinkData(data);
+
+		if (cleanedData['tags']) {
+			cleanedData['tags'] = goog.json.serialize(cleanedData['tags']);
+		}
+
+		if (cleanedData['data']) {
+			cleanedData['metadata'] = cleanedData['data'];
+			delete cleanedData['data'];
+		}
+
+		if (options['open_app']) {
+			cleanedData['open_app'] = true;
+		}
+
+		var referringLink = self._referringLink();
+		if (referringLink && !options['make_new_link']) {
+			cleanedData['link_click_id'] = referringLink.substring(
+				referringLink.lastIndexOf('/') + 1, referringLink.length
+			);
+		}
+
+		this._api(resources.deepview, cleanedData, function(err, data) {
+			if (err) {
+				return done(err);
+			}
+
+			if (typeof data === 'function') {
+				self._deepviewCta = data;
+			}
+			else {
+				self._server.createScript(data, true);
+				done(null);
+			}
+		});
+	}
+);
+
+/**
+ * @function Branch.deepviewCta
+ *
+ * Perform the branch deepview CTA (call to action) on mobile. Namely, depends on how
+ * *branch.deepview* is set up, the mobile users are redirected accordingly. If the deepview is
+ * configured with the option *`open_app`* being true, an immediate attempt is made as soon as
+ * deepview finishes, and thus the CTA is to visit the platform-appropriate app stores; if on the
+ * other hand, the deepview is configured with the option *`open_app`* being false, the CTA is to
+ * try to open app, and to visit the platform-appropriate app stores if the open-app attempt failes.
+ *
+ * If *branch.deepview* has not been called, an error will arise with a reminder to call
+ * *branch.deepview* first.
+ *
+ * ##### Usage
+ * ```js
+ * branch.deepviewCta();
```

Comment:
we may want to use an example with a link, since the idea is to put this behind a link. for example,

```
// If you're using jQuery, then:
$('a.deepview-cta').click(function() { branch.deepviewCta(); return false });
```

Now that I think of it, we may want to allow users to bind directly to a link like this:
```
$('a.deepview-cta').click(branch.deepviewCta);
```

or 

```
document.getElementById('my-elem').onClick = branch.deepviewCta;
```

in which case we'll either want to have deepviewCta take the click event as an argument, and call `.preventDefault()`, or we'll want to return false (o maybe both).

Though we shouldn't require it to be called like that - we should allow people to call it whenever they want. So we need to verify that preventDefault is callable and such.